benchmarks/bvh_driver/bvh_driver.cpp

```diff
@@ -328,42 +332,105 @@ int main(int argc, char *argv[])
     return 1;
   }
 
+  if (vm.count("exact-spec") > 0)
+  {
+    for (std::string option :
+         {"values", "queries", "predicate-sort", "neighbors", "buffer",
+          "source-point-cloud-type", "target-point-cloud-type"})
+    {
+      if (!vm[option].defaulted())
+      {
+        std::cout << "Conflicting options: \"exact-spec\" and \"" << option
+                  << "\", exiting..." << std::endl;
+        return EXIT_FAILURE;
+      }
+    }
+  }
+
   benchmark::Initialize(&pass_further.argc(), pass_further.argv());
   // Throw if some of the arguments have not been recognized.
   std::ignore =
       bpo::command_line_parser(pass_further.argc(), pass_further.argv())
           .options(bpo::options_description(""))
           .run();
 
-  // Google benchmark only supports integer arguments (see
-  // https://github.com/google/benchmark/issues/387), so we map the string to
-  // an enum.
   std::map<std::string, PointCloudType> to_point_cloud_enum;
   to_point_cloud_enum["filled_box"] = PointCloudType::filled_box;
   to_point_cloud_enum["hollow_box"] = PointCloudType::hollow_box;
   to_point_cloud_enum["filled_sphere"] = PointCloudType::filled_sphere;
   to_point_cloud_enum["hollow_sphere"] = PointCloudType::hollow_sphere;
-  int source_point_cloud_type = to_point_cloud_enum.at(source_pt_cloud);
-  int target_point_cloud_type = to_point_cloud_enum.at(target_pt_cloud);
+
+  PointCloudType source_point_cloud_type;
+  PointCloudType target_point_cloud_type;
+
+  auto label_construction = [&](std::string const &tree_name) -> std::string {
+    std::string s = std::string("BM_construction<") + tree_name + ">";
+    for (auto &var : {n_values, (int)source_point_cloud_type})
+      s += "/" + std::to_string(var);
+    return s.c_str();
+  };
+  auto label_knn_search = [&](std::string const &tree_name) -> std::string {
+    std::string s = std::string("BM_knn_search<") + tree_name + ">";
+    for (auto &var :
+         {n_values, n_queries, n_neighbors, sort_predicates_int,
+          (int)source_point_cloud_type, (int)target_point_cloud_type})
+      s += "/" + std::to_string(var);
+    return s.c_str();
+  };
+  auto label_radius_search = [&](std::string const &tree_name) -> std::string {
+    std::string s = std::string("BM_radius_search<") + tree_name + ">";
+    for (auto &var :
+         {n_values, n_queries, n_neighbors, sort_predicates_int, buffer_size,
+          (int)source_point_cloud_type, (int)target_point_cloud_type})
+      s += "/" + std::to_string(var);
+    return s.c_str();
+  };
+
+  if (vm.count("exact-spec") == 0)
+  {
+    exact_specs.resize(1);
+    auto &spec = exact_specs[0];
+    spec = std::to_string(n_values);
+    for (auto &var :
+         {n_queries, n_neighbors, sort_predicates_int, buffer_size,
+          (int)source_point_cloud_type, (int)target_point_cloud_type})
+      spec += "/" + std::to_string(var);
+  }
```

Comment:
This creates a dummy `a/b/c` from existing variables when no exact specs are given.